modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java

```diff
@@ -182,18 +192,32 @@ protected void doServeResource(
 		}
 	}
 
-	private JSONArray _getAssetCategoriesJSONArray(
-		ContentDashboardItem contentDashboardItem, Locale locale) {
+	private HashMap<String, Object> _buildVocabularyDetails(
+			Locale locale, AssetCategory assetCategory,
+			AssetVocabulary currentVocabulary)
+		throws PortalException {
 
-		List<AssetCategory> assetCategories =
-			contentDashboardItem.getAssetCategories();
+		return HashMapBuilder.<String, Object>put(
+			"categories",
+			() -> {
+				ArrayList<String> categoriesArray = new ArrayList<>();
```

Comment:
The type of the variable should be the interface List, instead the implementation class ArrayList.

Following Liferay style book, its name should be categories.